### PR TITLE
Add ACL and redirect for api.turbovote.org/users

### PR DESCRIPTION
**Pull request by ericnormand**
It should redirect to /user-http-api on one of the wildflies.

That is

```
http://api.turbovote.org/users/stuff => http://wildfly1/user-http-api/stuff
```

[Pivotal Card](https://www.pivotaltracker.com/story/show/99939626)

### CODE REVIEW

File: conf.json [hunk 1]
```diff
@@ -99,7 +99,9 @@
         "acl is_turbovote_staging hdr_end(host) -i turbovote.net",
         "use_backend turbovote_staging if is_turbovote_staging",
         "acl is_online_voter_reg_api hdr_end(host) -i online-voter-reg-api.turbovote.org",
-        "use_backend online_voter_reg_api if is_online_voter_reg_api"
+        "use_backend online_voter_reg_api if is_online_voter_reg_api",
+        "acl is_user_http_api base -i ^api[.]turbobote[.]org/users.*",
```
Comment (cap10morgan) [inline conf.json]: Would this be faster?

`acl is_user_http_api base_beg -i api.turbovote.org/users`

File: conf.json [hunk 2]
```diff
@@ -99,7 +99,9 @@
         "acl is_turbovote_staging hdr_end(host) -i turbovote.net",
         "use_backend turbovote_staging if is_turbovote_staging",
         "acl is_online_voter_reg_api hdr_end(host) -i online-voter-reg-api.turbovote.org",
-        "use_backend online_voter_reg_api if is_online_voter_reg_api"
+        "use_backend online_voter_reg_api if is_online_voter_reg_api",
+        "acl is_user_http_api base_beg api.turbovote.org/users",
+        "use_backend user_http_api if is_user_http_api"
       ],
```
Comment (ericnormand) [inline conf.json]: So, I'm getting this error when I run it in docker:

```
I, [2015-08-11T14:58:23.306406 #5]  INFO -- Synapse::Synapse: synapse: starting...
I, [2015-08-11T14:58:23.312923 #5]  INFO -- Synapse::Synapse: synapse: regenerating haproxy config
W, [2015-08-11T14:58:23.313235 #5]  WARN -- Synapse::Haproxy: synapse: no backends found for watcher wildfly
W, [2015-08-11T14:58:23.313428 #5]  WARN -- Synapse::Haproxy: synapse: no backends found for watcher turbovote
[ALERT] 222/145823 (16) : parsing [/etc/haproxy/haproxy.cfg:37] : error detected while parsing ACL 'is_user_http_api'.
[ALERT] 222/145823 (16) : parsing [/etc/haproxy/haproxy.cfg:38] : error detected while parsing switching rule.
[ALERT] 222/145823 (16) : Error(s) found in configuration file : /etc/haproxy/haproxy.cfg
[ALERT] 222/145823 (16) : Fatal errors found in configuration.
E, [2015-08-11T14:58:23.329883 #5] ERROR -- Synapse::Synapse: synapse: encountered unexpected exception #<RuntimeError: failed to reload haproxy via service haproxy reload:  * Reloading haproxy haproxy
   ...fail!> in main thread
W, [2015-08-11T14:58:23.330039 #5]  WARN -- Synapse::Synapse: synapse: exiting; sending stop signal to all watchers
I, [2015-08-11T14:58:23.330227 #5]  INFO -- Synapse::DockerWatcher: synapse: stopping watcher wildfly using default stop handler
I, [2015-08-11T14:58:23.330306 #5]  INFO -- Synapse::DockerWatcher: synapse: stopping watcher turbovote using default stop handler
/var/lib/gems/1.9.1/bundler/gems/synapse-946eb22493d4/lib/synapse/haproxy.rb:778:in `restart': failed to reload haproxy via service haproxy reload:  * Reloading haproxy haproxy (RuntimeError)
   ...fail!
	from /var/lib/gems/1.9.1/bundler/gems/synapse-946eb22493d4/lib/synapse/haproxy.rb:550:in `update_config'
	from /var/lib/gems/1.9.1/bundler/gems/synapse-946eb22493d4/lib/synapse.rb:51:in `block in run'
	from /var/lib/gems/1.9.1/bundler/gems/synapse-946eb22493d4/lib/synapse.rb:43:in `loop'
	from /var/lib/gems/1.9.1/bundler/gems/synapse-946eb22493d4/lib/synapse.rb:43:in `run'
	from /var/lib/gems/1.9.1/bundler/gems/synapse-946eb22493d4/bin/synapse:60:in `<top (required)>'
	from /var/lib/gems/1.9.1/bin/synapse:23:in `load'
	from /var/lib/gems/1.9.1/bin/synapse:23:in `<main>'
```

Comment (ericnormand) [inline conf.json]: More drama. If I define the acl as:
```
acl is_online_voter_reg_api hdr_end(host) -i online-voter-reg-api.turbovote.org
```

It works.

Comment (ericnormand) [inline conf.json]: It seems that `base_beg` doesn't work. But neither does`base` or `uri`.

Comment (ericnormand) [inline conf.json]: Ha! `url` works.

**Comment by cap10morgan** [comment]
LGTM

**Pull request merged**